st2api/st2api/controllers/resource.py

```diff
@@ -187,6 +198,16 @@ def _get_by_name(self, resource_name):
 
         return resource_db
 
+    def _get_from_model_kwargs_for_request(self, request):
+        """
+        Retrieve kwargs which are passed to "LiveActionAPI.model" method.
+
+        :param request: Pecan request object.
+
+        :rtype: ``dict``
+        """
+        return self.from_model_kwargs
```

Comment:
I am actually confused with this method. It takes in `request` argument but does nothing with it :P. Can you use abc and declare it an abstract method and have a pass implementation here? I also see that you set the from_model_kwargs in executions controller but it's kind of confusing. What you want is a default kwargs defined for the controller and then some kv pairs you get from requests. This method should return a fully formed dict from these two dicts.